sqlair.go

```diff
@@ -96,20 +96,28 @@ func (q *Query) Run() error {
 	if q.err != nil {
 		return q.err
 	}
-	_, err := q.qs.ExecContext(q.ctx, q.qe.QuerySQL(), q.qe.QueryArgs()...)
+	_, err := q.exec()
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func (q *Query) exec() (sql.Result, error) {
+	return q.qs.ExecContext(q.ctx, q.qe.QuerySQL(), q.qe.QueryArgs()...)
+}
+
+func (q *Query) query() (*sql.Rows, error) {
+	return q.qs.QueryContext(q.ctx, q.qe.QuerySQL(), q.qe.QueryArgs()...)
```

Comment:
It is [generally discouraged](https://go.dev/blog/context-and-structs) to store a context as a struct member.

There is no guard against calling `Run` twice, which could result in unpredictable behaviour.

I would prefer for methods that materialise an action against the database, whether DML or SQL, to accept a context.

Reply:
Good point, I've added a PR (#67) to move the context to the `Get/Run/GetAll` stage.